hs_core/hydroshare/resource.py

```diff
@@ -559,13 +559,20 @@ def create_new_version_resource(ori_res, new_res):
         res_avail_date = new_res.metadata.dates.all().filter(type='available')[0]
         new_res.metadata.create_element('date', type='available', start_date=res_avail_date.start_date, end_date=res_avail_date.end_date)
 
-    # add Relation element to link source and target resources
+    # add or update Relation element to link source and target resources
     if new_res.metadata.identifiers.all().filter(name="hydroShareIdentifier"):
         hs_identifier = new_res.metadata.identifiers.all().filter(name="hydroShareIdentifier")[0]
         ori_res.metadata.create_element('relation', type='isReplacedBy', value=hs_identifier.url)
     else:
         ori_res.metadata.create_element('relation', type='isReplacedBy', value=new_res.short_id)
```

Comment:
@dtarb @horsburgh Here is the content on workflow creating a new version of a published resource in the design spec: ```Users can create a new version of a resource even if the previous version has been published and has been assigned a DOI. Editing the content of a formally published resource constitutes creating a new version. Only the most recent version of a resource in an obsolescence chain will be “active” and discoverable using HydroShare search services, but linkage to previous versions will be maintained and provided via the HydroShare web user interface.  The landing page for the most recent version of a resource will provide links back to previous versions. DOI’s for obsoleted versions will resolve to the landing page for the obsoleted version, which will provide the archived data and metadata for that version as well as a notice that a newer version is available and a link to the most recent version.
 
If a user wants to create a new version of a published resource, the process is exactly the same as that used for new versions of unpublished resources (see above).```
As you can see, we follow the same workflow for creating a new version of a published or unpublished resource. Also, before the user clicks the versioning button, we do inform users front and center that the obsoleted version will be removed from searches and My Resources page in HydroShare and that if they want the new version to be discoverable, they need to make it public or publish it. I will leave this to you to work out the policy adjustment if needed, and you can create new issues if we need to change anything, but let's get this fix PR merged regardless since it not only made cosmetic changes as suggested by @horsburgh, but also fixed an bug which prevents users from creating obsolescence chain that includes more than 2 resources, as well as updated original resource's updated date metadata element properly.